sno/structure.py

```diff
@@ -87,15 +87,15 @@ def version(self):
         """Returns the dataset version to use for this entire repo."""
         return get_structure_version(self.repo, self.tree)
 
-    def decode_path(self, path):
+    def decode_path(self, full_path):
         """
-        Given a path in the sno repository - eg "table_A/.sno-table/49/3e/Bg==" -
+        Given a path in the sno repository - eg "path/to/dataset/.sno-table/49/3e/Bg==" -
         returns a tuple in either of the following forms:
-        1. (table, "feature", primary_key)
-        2. (table, "meta", metadata_file_path)
+        1. (dataset_path, "feature", primary_key)
+        2. (dataset_path, "meta", meta_item_path)
         """
-        table, table_path = path.split("/.sno-table/", 1)
-        return (table,) + self.get(table).decode_path(table_path)
+        dataset_path, rel_path = re.split("(?=/.sno-table/)", full_path, 1)
```

Comment:
should pre-compile this regex outside the function body. Also it should be a raw string

Reply:
Timed this - precompiling a regex is 25% faster, using simple string split and then re-prepending it is 75% faster. Using simple string operations now. No guarantee that it makes any difference to the code overall, but no harm in it.